frontend/src/bundles/talent-onboarding/talent-onboarding-api.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    ApiPath,
+    ContentType,
+    UserDetailsApiPath,
+} from 'shared/build/index.js';
+
+import { HttpApiBase } from '~/framework/api/api.js';
+import { type Http } from '~/framework/http/http.js';
+import { type Storage } from '~/framework/storage/storage.js';
+
+import { type UserDetailsGeneralCustom } from './types/types.js';
+
+type Constructor = {
+    baseUrl: string;
+    http: Http;
+    storage: Storage;
+};
+
+class TalentOnBoardingApi extends HttpApiBase {
+    public constructor({ baseUrl, http, storage }: Constructor) {
+        super({ path: ApiPath.USER_DETAILS, baseUrl, http, storage });
+    }
+
+    public async getUserDetailsByUserId(
+        payload: Partial<UserDetailsGeneralCustom>,
+    ): Promise<UserDetailsGeneralCustom | null> {
+        const { userId = '' } = payload;
+
+        const response = await this.load(
+            this.getFullEndpoint(UserDetailsApiPath.ROOT, userId, {}),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+        return await response.json<UserDetailsGeneralCustom>();
```

Comment:
Don't use return await (http://hassansin.github.io/Why-return-await-Is-a-Bad-Idea)
```suggestion
        return response.json<UserDetailsGeneralCustom>();
```

Reply:
Hello, this is based off of an earlier version of #344. This has been addressed there in a newer version.